cli/src/lib/zk-sync-era-diff.ts

```diff
@@ -93,6 +94,44 @@ export class ZkSyncEraDiff {
     await this.writeFacets(filter, baseDirOld, baseDirNew);
     await this.writeVerifier(filter, baseDirOld, baseDirNew, l1Client);
     await this.writeSystemContracts(filter, baseDirOld, baseDirNew, l2Client, github, ref);
+    await this.writeSpecialContracts(baseDirNew, github, ref);
+  }
+
+  private async writeSpecialContracts(dir: string, github: GithubClient, ref: string) {
+    const baseDirAA = path.join(dir, "defaultAA");
+    const baseDirBL = path.join(dir, "bootloader");
+
+    const rawHashes = await github.downloadFile("system-contracts/SystemContractsHashes.json", ref);
+    const hashes = systemContractHashesSchema.parse(JSON.parse(rawHashes));
+
+    if (this.newAA !== ZERO_U256) {
+      const defaultAccountHash = hashes.find((h) => h.contractName === "DefaultAccount");
+      if (!defaultAccountHash || defaultAccountHash.bytecodeHash !== this.newAA) {
+        throw new Error(`Default Account contract byte code hash does not match in ref: ${ref}`);
+      }
+
+      const sourcesAA = await github.downloadContract("DefaultAccount", ref);
+      const contractAA = new ContractData("DefaultAA", sourcesAA, ADDRESS_ZERO);
+      await contractAA.writeSources(baseDirAA);
```

Comment:
Intermediary creating the intermediary contract data object to use the logic to write source code.